From eb5308b7868c5dfda080f4753656ee62a057ff41 Mon Sep 17 00:00:00 2001
From: Simon Hughes <simon.hughes@arm.com>
Date: Fri, 22 Jun 2018 10:39:57 +0100
Subject: [PATCH] IOTMBL-438: Fix optee_test error: 'memmove'
 -Werror=array-bounds.

The following provides more information on this commit:
- The reported error was "error: 'memmove' pointer overflow between offset
  [0, 8] and size [4294967295, 2147483647] accessing array 'tmp' with
  type 'uint8_t[1024]' {aka 'unsigned char[1024]'} [-Werror=array-bounds].
- The error is caused by using GCC 8.1.1.

Signed-off-by: Simon Hughes <simon.hughes@arm.com>
---
 host/xtest/regression_4000.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/host/xtest/regression_4000.c b/host/xtest/regression_4000.c
index ebc9bae..8b10c55 100644
--- a/host/xtest/regression_4000.c
+++ b/host/xtest/regression_4000.c
@@ -5041,6 +5041,7 @@ static void xtest_tee_test_4011(ADBG_Case_t *c)
 			if (tmp[m] != 0xff)
 				break;
 		/* 4.2 */
+		tmp_size = sizeof(tmp);
 		memmove(tmp + n + i, tmp + m, tmp_size - m);
 		/* 4.3 */
 		for (n = n + tmp_size - m + i; n < tmp_size; n++)
-- 
2.7.4

